### PR TITLE
Add function call request param

### PR DESCRIPTION
**Pull request by DaviRain-Su**
close: #72 
#76 

- Remove example/Cargo.toml, add to root Cargo.toml
- Add function-call example get_current_weather

### CODE REVIEW

**Review by b0xtch — APPROVED** [review]

**Comment by b0xtch** [comment]
I would also update the readme to add function as feature

**Comment by DaviRain-Su** [comment]
I am currently writing an example code for function calls, but I encountered an issue with the error: `JSONDeserialize(Error("invalid type: null, expected a string", line: 11, column: 23)).`  I need some help to resolve this.

**Comment by 64bit** [comment]
Hi @DaviRain-Su, Thank you for the PR and adding example for this new feature - its a great way to test it.

Here are some hints for you to debug this further:
- see the updated upstream OpenAPI spec and match it with Rust type in code
- setup tracing_subscriber so that `tracing::error!( ... )` can log the actualy json payload causing deserializtion error. `examples/rate-limit-completions` has an example - however you might need to change log level. 

Hope this helps

**Comment by DaviRain-Su** [comment]
Successfully added an example that can use function calls.

File: async-openai/src/types/types.rs [hunk 1]
```diff
@@ -698,17 +699,39 @@ pub enum Role {
 pub struct ChatCompletionRequestMessage {
     /// The role of the author of this message.
     pub role: Role,
+    #[serde(skip_serializing_if = "Option::is_none")]
     /// The contents of the message
-    pub content: String,
+    pub content: Option<String>,
```
Comment (64bit) [inline async-openai/src/types/types.rs:704]: Comment needs to be updated (please see latest spec)

> The contents of the message. `content` is required for all messages except assistant messages with function calls.

**Review by 64bit — CHANGES_REQUESTED** [review]
It appears that naming does not adheres to OpenAPI spec.

For example  `ChatCompletionFunctions` , Please see the names of new types in this PR https://github.com/64bit/async-openai/pull/73


For background: all types in `types.rs` are named based on OpenAPI component schema.

**Comment by DaviRain-Su** [comment]
Have fix all.

**Comment by 64bit** [comment]
`function_call` in `ChatCompletionRequestMessage` and `ChatCompletionResponseMessage` can also be a struct with `name` and `arguments` - I'll take care of it before publishing.

Thank you for addressing the comments,  nice work on the new example!

**Review by 64bit — APPROVED** [review]

**Comment by 64bit** [comment]
This is released in v0.12.0

**Pull request merged**